wasm-host/xrpld-number/c/src/number_c.h

```diff
@@ -0,0 +1,95 @@
+#ifndef XRPLD_NUMBER_C_H
+#define XRPLD_NUMBER_C_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <stddef.h>
+
+// Opaque pointer to the C++ Number object
+typedef struct Number Number;
+
+// Error codes
+typedef enum {
+    NUMBER_SUCCESS = 0,
+    NUMBER_ERROR_OVERFLOW = 1,
+    NUMBER_ERROR_DIVIDE_BY_ZERO = 2,
+    NUMBER_ERROR_INVALID_ARGUMENT = 3,
+    NUMBER_ERROR_OUT_OF_MEMORY = 4,
+    NUMBER_ERROR_UNKNOWN = 5
+} NumberError;
+
+// Rounding modes
+typedef enum {
+    ROUNDING_TO_NEAREST = 0,
+    ROUNDING_TOWARDS_ZERO = 1,
+    ROUNDING_DOWNWARD = 2,
+    ROUNDING_UPWARD = 3
+} RoundingMode;
+
+// Construction and destruction
+Number* number_new();
+Number* number_new_from_int64(int64_t mantissa, NumberError* error);
```

Comment:
No number from uin64/uint32 support?

Reply:
Number not, but host-functions does. Maybe you should  add uint support in other PR